core/modules/layout/layout.context.admin.inc

```diff
@@ -133,15 +133,15 @@ function layout_context_add_form($form, &$form_state, Layout $layout = NULL, $co
 /**
  * Form callback; Displays form for adding new relationships to a layout.
  *
- * @param Layout $layout
+ * @param ?Layout $layout
  *   The layout to which a context is being added.
  * @param int|null $relationship_id
  *   The ID of the relationship being configured. If adding a new
  *   relationship, no value will be passed in.
  *
  * @ingroup forms
  */
-function layout_context_relationship_add_form($form, &$form_state, Layout $layout = NULL, $relationship_id = NULL) {
+function layout_context_relationship_add_form($form, &$form_state, ?Layout $layout, $relationship_id = NULL) {
```

Comment:
Similar to above - could Layout ever be null when adding a relationship?

The "NULL" assignment has been stripped, BTW, and the doc block uses nullable type notation. Is that an oversight, or by intention? (Using nullable type notation like that is OK for doc block, btw, so that's not my concern here.)

Reply:
`?Layout $layout = NULL` would be backward compatible with `Layout $layout = NULL` for code that calls `layout_context_relationship_add_form()` as `layout_context_relationship_add_form($form, $form_state)`. `layout_context_relationship_add_form($form, &$form_state, Layout ?$layout, $relationship_id = NULL)` requires at least three arguments, since the third one is not optional, even if the function could be called as `layout_context_relationship_add_form($form, $form_state, NULL)`.

Reply:
(I wonder too if it makes sense to pass a `NULL` layout. This should be changed in a different issue, though, as this issue should make the code compatible with PHP 8.4.)

Reply:
Thanks, added `= NULL` even though it probably should be required. Separate issue.

Reply:
I changed my mind here and went ahead and made `$layout` a required and non-nullable parameter here.